test/datagateway_api/icat/test_icat_client.py

```diff
@@ -10,6 +15,47 @@ def test_init(self):
 
         assert not test_icat_client.autoLogout
 
+    @pytest.mark.parametrize(
+        "client_use, expected_url, expected_check_cert",
+        [
+            pytest.param(
+                "datagateway_api",
+                "https://localhost:8181/ICATService/ICAT?wsdl",
+                False,
+                id="DataGateway API Usage",
+            ),
+            pytest.param(
+                "search_api",
+                "https://localhost.testdomain:8181/ICATService/ICAT?wsdl",
+                True,
+                id="Search API Usage",
+            ),
+        ],
+    )
+    def test_client_use(
+        self, test_config_data, client_use, expected_url, expected_check_cert,
+    ):
+        with patch("builtins.open", mock_open(read_data=json.dumps(test_config_data))):
+            api_config = APIConfig.load("test/path")
+
+            class MockClient:
+                def __init__(url, checkCert=True):  # noqa
+                    print(f"URL: {url}, Cert: {checkCert}")
+                    # Would've preferred to assign these values to self but this didn't
+                    # seem to be possible
+                    Client.url = f"{url}/ICATService/ICAT?wsdl"
+                    Client.checkCert = checkCert
+
+            with patch(
+                "datagateway_api.src.common.config.Config.config", api_config,
+            ):
+                with patch(
+                    "icat.client.Client.__init__", side_effect=MockClient.__init__,
+                ):
+                    test_icat_client = ICATClient(client_use)
+                    assert test_icat_client.url == expected_url
+                    assert test_icat_client.checkCert == expected_check_cert
+
```

Comment:
Is it not possible to make use of `test_config` here? Is the print statement needed in the `__init__`?

```suggestion
    def test_client_use(
        self, test_config, client_use, expected_url, expected_check_cert,
    ):
        class MockClient:
            def __init__(url, checkCert=True):  # noqa
                print(f"URL: {url}, Cert: {checkCert}")
                # Would've preferred to assign these values to self but this didn't
                # seem to be possible
                Client.url = f"{url}/ICATService/ICAT?wsdl"
                Client.checkCert = checkCert

        with patch(
            "datagateway_api.src.common.config.Config.config", test_config,
        ):
            with patch(
                "icat.client.Client.__init__", side_effect=MockClient.__init__,
            ):
                test_icat_client = ICATClient(client_use)
                assert test_icat_client.url == expected_url
                assert test_icat_client.checkCert == expected_check_cert

```